Assets/_BForBoss/_Core/Scripts/HUD/CountdownViewBehaviour.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using Perigon.Utility;
+using TMPro;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class CountdownViewBehaviour : MonoBehaviour
+    {
+        [SerializeField, Resolve] private TMP_Text _countdownLabel;
+        private float _time;
+        private bool _isRunning;
+
+        private float _amountOfTime;
+        private Action _onCountdownComplete;
+
+        public void InitializeCountdown(float amountOfTime, Action onCountdownComplete)
+        {
+            _amountOfTime = amountOfTime;
+            _time = amountOfTime;
+            _onCountdownComplete = onCountdownComplete;
+        }
+
+        public void StartCountdown()
+        {
+            gameObject.SetActive(true);
+            _isRunning = true;
+        }
+
+        public void PauseCountdown()
+        {
+            _isRunning = false;
+        }
+
+        public void Reset()
+        {
+            _time = _amountOfTime;
+            SetTimerLabel(_amountOfTime);
+        }
+
+        private void CompleteCountdown()
+        {
+            _onCountdownComplete?.Invoke();
+            gameObject.SetActive(false);
+            Reset();
+        }
+
+        private void Update()
+        {
+            if (!_isRunning)
+            {
+                return;
+            }
+
+            if (_time >= 0.0f)
+            {
+                _time -= Time.deltaTime;
+                SetTimerLabel(_time);
+            }
+            else
+            {
+                _isRunning = false;
```

Comment:
1. you can place `_isRunning = false;` within `CompleteCountDown`
2. You can put `SetTimerLabel(_time);` at the end of the Update outside of these if statements.

Reply:
Since CompleteCountDown sets the GameObject to false, Players wont actually see The timer when it hits 0, So I purposely put the `SetTimerLabel(_time)` twice. 

What Ill do is add this to the `CompleteCountDownMethod` as well